core/src/utils/overlays.ts

```diff
@@ -612,3 +612,45 @@ export const safeCall = (handler: any, arg?: any) => {
 };
 
 export const BACKDROP = 'backdrop';
+
+/**
+ * Constructs a trigger interaction for an overlay.
+ * Presents an overlay when the trigger is clicked.
+ *
+ * Usage:
+ * ```ts
+ * triggerController = createTriggerController();
+ * triggerController.configureTriggerInteraction(el, trigger);
+ * ```
+ */
+export const createTriggerController = () => {
```

Comment:
I'm guessing other components could be adjusted to use this as well? (Maybe better for a separate PR?)

Reply:
Correct. I think the only exception is `ion-popover`, since it has additional trigger actions/logic that wouldn't make sense being abstracted into this. 

But `ion-modal` could be refactored to this, if we agree with the implementation pattern, and the remaining inline overlays could benefit from this.